src/store/chatStore.ts

```diff
@@ -398,33 +412,143 @@ const sendFile = async (chatId, selectedFile, isBlob = false, isRecording = fals
         });
-        console.log('File uploaded.');
-    } catch (e) {
-        let errorBody = '';
-        if (e.message == 'Request failed with status code 413') {
-            errorBody = 'ERROR: File exceeds 100MB limit!';
-        } else {
-            errorBody = 'ERROR: File failed to send!';
-        }
-
-        const failedMsg = {
-            ...msgToSend,
-            body: errorBody,
-            type: 'STRING',
-        };
-        addMessage(chatId, failedMsg);
+            } catch (e) {
+        sendFileCatch(e,uuid)
     }
 };
 
+const sendFileCatch = (e,uuid) => {
+    let errorBody = '';
+    if (e.message == 'Request failed with status code 413') {
+        //!TODO Upload limit
+        errorBody = 'ERROR: File exceeds 20MB limit!';
+        imageUploadQueue.value = imageUploadQueue.value.map(el => {
+            if (uuid === el.id) {
+                return { ...el, error_message: 'File exceeds 20MB limit!', error: true };
+            }
+            return el;
+        });
+    } else {
+
+        errorBody = 'ERROR: File failed to send!';
+        imageUploadQueue.value = imageUploadQueue.value.map(el => {
+            if (uuid === el.id) {
+                return { ...el, error_message: 'File failed to send', error: true, retry: true };
+            }
+            return el;
+        });
+
+    }
+}
+
+export const retrySendFile = async (file) => {
+    if(!file) return;
+
+    const {id:uuid, chatId, selectedFile} = file
+
+    let formData = new FormData();
+
+    formData.append('file', selectedFile);
+
+    //When a upload fails in chat and you retry
+    const CancelToken = axios.CancelToken;
+    const source = CancelToken.source();
+
+
+    try {
+        imageUploadQueue.value.map(el => {
+            if(el.id === uuid){
+                return {
+                    ...el,
+                    error_message: '',
+                    loaded: 0,
+                    retry: false,
+                    error: false,
+                    cancelToken: source
+                }
+            }
+            return el
+        })
+
+
+        await axios.post(`${config.baseUrl}api/files/${chatId}/${uuid}`, formData, {
+            headers: {
+                'Content-Type': 'multipart/form-data',
+            },
+            cancelToken: source.token,
+            onUploadProgress: ({ loaded: progress, total }) => {
+                imageUploadQueue.value = imageUploadQueue.value.map(item => {
+
+                    if (item.id === uuid) {
+                       return {
+                            ...item,
+                            loaded: progress,
+                            total: total,
+                            retry: false,
+                           error: false,
+                           error_message: ""
+                        };
+
+                    }
+                    return item;
+                });
+            },
+        });
+    } catch (e) {
+        sendFileCatch(e, uuid)
```

Comment:
naming is not good, a function name should represent what its doing not from where it comes